mkdocs/commands/serve.py

```diff
@@ -74,6 +78,12 @@ def error_handler(code):
                 for d in config['theme'].dirs:
                     server.watch(d)
 
+            # convert to set in case CLI params match config file settings
+            watch_list = set(config["watch"])
```

Comment:
`set` has non-deterministic order. You might use `dict.fromkeys` instead of `set`.

Reply:
Ordering isn't important indeed, but I always have a hunch that it'll cause some bug that appears only randomly. Why do things non-deterministically if you can just as easily do deterministically.

```suggestion
            watch_list = dict.fromkeys(config["watch"])
```